### PR TITLE
Anirud/sanity check script + Edits to install_tt_buda.md

### PR DESCRIPTION
**Pull request by anirudTT**
This PR adds the following: 

- Sanity Checker Script 
  - added the script and its placed within the first_5_steps folder
  - instructions for how to use the script have been added within the 1_install_tt_buda.md markdown file

- Edits to 1_install_tt_buda.md
   - some edits to better explain install of [tt-kmd](https://github.com/tenstorrent/tt-buda-demos/commit/cdf5417d5f04bf6ede29d14efe147275267f8f53#:~:text=*Pro%2DTip%3A%20ensure%20that%20you%20are%20within%20the%20home%20directory%20of%20the%20local%20clone%20version%20of%20%5Btt%2Dkmd%5D(https%3A//github.com/tenstorrent/tt%2Dkmd)%20when%20performing%20the%20installation%20steps*) because I noticed other people having the same error on a slack channel
   - The install instructions have been updated to explicitly direct users to the README file of the specific utility they intend to install.
   - Table of contents have been modified 
      - installation instructions are in thier own section
      - smoke test and the new sanity script are under a section called **Verifications**
   - Some opinionated appearance changes to the markdown file -- in terms of adding dividers 

- Edits to the .gitignore file 
   - to ensure it now ignore all files with the extension whl
   - ignore mac os generated files like  .DS store

### CODE REVIEW

File: .gitignore
Comment (milank94) [inline .gitignore:1]: Add a newline to the end of file.

File: first_5_steps/1_install_tt_buda.md [hunk 1]
```diff
@@ -53,28 +56,32 @@ sudo reboot
...
 apt-get install -y python3.8-venv libboost-all-dev libgoogle-glog-dev libgl1-mesa-glx libyaml-cpp-dev ruby
 apt-get install -y build-essential clang-6.0 libhdf5-serial-dev libzmq3-dev
 ```
 
 ### TT-SMI
 
-Please navigate to [tt-smi](https://github.com/tenstorrent/tt-smi).
+Please navigate to [tt-smi](https://github.com/tenstorrent/tt-smi) homepage and follow instructions within the README.
 
 ### Tenstorrent Software Package
```
Comment (milank94) [inline first_5_steps/1_install_tt_buda.md:86]: I think we should remove this section. This only applies for the Python Environment Installation, which is in the next section.

File: first_5_steps/1_install_tt_buda.md [hunk 2]
```diff
@@ -31,7 +31,9 @@ Once you have identified the release version you would like to install, you can
    6. [Tenstorrent Software Package](#tenstorrent-software-package)
```
Comment (milank94) [inline first_5_steps/1_install_tt_buda.md:31]: I think we should remove the Tenstorrent Software Package section (see other comment below).

And I think we should have a header that says something like "PyBuda Installation" and have the next two sections be subsections.

Something like this -- open to making it nicer:
```markdown
6. [PyBuda Installation]
  i. [Python Environment Installation]
  ii. [Docker Container Installation]
```

File: first_5_steps/sanity_check.sh [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+if command_exists "$TT_SMI_COMMAND" && "$TT_SMI_COMMAND" "$TT_SMI_HELP_ARG" >/dev/null 2>&1; then
+    echo -e "${GREEN}5. TT-SMI Installation: OK${NC}"
+else
+    echo -e "${RED}5. TT-SMI Installation: Not installed${NC}"
+    FAILED_CHECKS+=("5. TT-SMI Installation")
+fi
+CURRENT_STEP=$((CURRENT_STEP + 1))
+print_progress
+
+# 6. TT-Flash check
```
Comment (milank94) [inline first_5_steps/sanity_check.sh:118]: This looks like a duplicate of #3 Device Firmware Update

File: first_5_steps/sanity_check.sh [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+if command_exists lsmod && lsmod | grep "$LSMOD_GREP" >/dev/null 2>&1; then
+    echo -e "${GREEN}2. PCI Driver Installation: OK${NC}"
+else
+    echo -e "${RED}2. PCI Driver Installation: Not installed${NC}"
+    FAILED_CHECKS+=("2. PCI Driver Installation")
+fi
+CURRENT_STEP=$((CURRENT_STEP + 1))
+print_progress
+
+# 3. tt-flash check
```
[the remark this answers is not in the document]
Comment (milank94) [reply first_5_steps/sanity_check.sh:78]: For example, I ran on a system that I know has everything configured but I still get the errors:
image

File: .gitignore
Comment (anirudTT) [reply .gitignore:1]: added new line

**Pull request closed without merging**